### PR TITLE
Replace IntegerOrAuto with ZIndex

### PR DESCRIPTION
**Pull request by nox**
It's its only use.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/20079)

### CODE REVIEW

File: components/style/values/generics/position.rs [hunk 1]
```diff
@@ -24,3 +24,39 @@ impl<H, V> Position<H, V> {
...
+    /// Returns `auto`
+    #[inline]
+    pub fn auto() -> Self {
+        ZIndex::Auto
+    }
+
+    /// Returns whether `self` is `auto`.
+    #[inline]
+    pub fn is_auto(self) -> bool {
+        match self {
```
Comment (emilio) [inline components/style/values/generics/position.rs]: nit: `matches!`?

**Review by emilio — APPROVED** [review]

**Comment by nox** [comment]
Oops, I can't do negation.

@bors-servo r=emilio

**Comment by CYBAI** [comment]
```
{"status": "FAIL", "group": "default", "message": "child.document.querySelector(...) is null", "stack": "onload/<@http://web-platform.test:8000/html/browsers/browsing-the-web/navigating-across-documents/javascript-url-abort/javascript-url-abort-return-value-undefined.tentative.html:18:7\nTest.prototype.step_timeout/<@http://web-platform.test:8000/resources/testharness.js:1552:20\nTest.prototype.step@http://web-platform.test:8000/resources/testharness.js:1494:20\nTest.prototype.step_func/<@http://web-platform.test:8000/resources/testharness.js:1518:20\n", "subtest": "Not aborting fetch for javascript:undefined navigation", "test": "/html/browsers/browsing-the-web/navigating-across-documents/javascript-url-abort/javascript-url-abort-return-value-undefined.tentative.html", "line": 133269, "action": "test_result", "expected": "TIMEOUT"}
{"status": "OK", "group": "default", "message": null, "stack": null, "subtest": null, "test": "/html/browsers/browsing-the-web/navigating-across-documents/javascript-url-abort/javascript-url-abort-return-value-undefined.tentative.html", "line": 133270, "action": "test_result", "expected": "TIMEOUT"}
```

**Comment by jdm** [comment]
@bors-servo retry
* #20087

**Pull request merged**